notes/stv/20200124-hdfs-disk-space-issue.txt

```diff
@@ -0,0 +1,114 @@
+#
+# <meta:header>
+#   <meta:licence>
+#     Copyright (c) 2019, ROE (http://www.roe.ac.uk/)
+#
+#     This information is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU General Public License as published by
+#     the Free Software Foundation, either version 3 of the License, or
+#     (at your option) any later version.
+#
+#     This information is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU General Public License for more details.
+#  
+#     You should have received a copy of the GNU General Public License
+#     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   </meta:licence>
+# </meta:header>
+#
+#
+
+## HDFS Disk space issue
+## Nigel's noticed an issue when trying to running the following Spark job
+
+%spark.pyspark
+# Q: why am I having to prefix interpreter bindings with "spark." all of a sudden ?!
+
+# define the data frame source on the given column selection only (we don't want to read the whole thing ... presumably?)
+df = sqlContext.read.parquet("/hadoop/gaia/parquet/gdr2/gaia_source/*.parquet").select("source_id")
+
+# register as an sql context queryable object
+sqlContext.registerDataFrameAsTable(df, "gaia_source_ids")
+
+
+## Looks like a disk space issue
+## Exception message in Zeppelin cell: "No space left on device"
+
+
```

Comment:
Create a minimal Spark job that replicates the issue, run it from the command line and log the error message.